config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,81 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+// BGPPeer describes a BGP peer.
+type BGPPeer struct {
+	// asn is the Autonomous System Number of the peer.
+	//
+	// +optional
+	ASN string `json:"asn,omitempty"`
+
+	// ip is the IP address of the peer, as reachable from the cluster.
+	// It may be either IPv4 or IPv6.
+	//
+	// +kubebuilder:validation:Required
+	// + ---
+	// + {IPv4 or IPv6} validation in crd.yaml-patch
+	IP string `json:"ip"`
+
+	// password for BGP authentication against the peer.
+	//
+	// +optional
+	Password string `json:"password,omitempty"`
```

Comment:
no confidential information in CRs. This must reference a secret in ns/openshift-config.

Reply:
ack, I'll work on this next. Thanks

Reply:
I'm not sure yet how to address this; since this whole thing is being consumed by a static pod which has no access to ConfigMaps or secrets for now. I'll investigate with @mdbooth on how we can achieve this.